nucypher/characters/base.py

```diff
@@ -124,12 +123,23 @@ def __init__(self,
             self._crypto_power = CryptoPower(power_ups=self._default_crypto_powerups)
 
         self._checksum_address = checksum_public_address
+
+        # Fleet and Blockchain Connection (Everyone)
+        if not domains:
+            domains = (NodeConfiguration.DEFAULT_DOMAIN, )
+
+        # Needed for on-chain verification
+        if not self.federated_only:
+            self.blockchain = blockchain or Blockchain.connect()
+            self.miner_agent = MinerAgent(blockchain=blockchain)
```

Comment:
This logic is somewhat weird, since all characters, even Enrico will get a miner agent. But for the moment I guess it's OK! 

Perhaps a TODO?

Reply:
https://github.com/nucypher/nucypher/issues/1041